library/agent/Agent.ts

```diff
@@ -609,4 +612,27 @@ export class Agent {
         return this.sendHeartbeatEveryMS;
     }
   }
+
+  private registerShutdownHandler() {
+    const shutdownEvents = ["SIGTERM", "SIGINT", "SIGBREAK", "SIGHUP"];
+
+    shutdownEvents.forEach((event) => {
+      process.on(event, () => {
```

Comment:
Does this result in node.js not exiting like default?

Reply:
Yes, that is why process.exit is called